packages/amplify-category-auth/resources/auth-custom-resource/hostedUIProviderLambda.js

```diff
@@ -12,17 +13,51 @@ async function handleEvent(event, context) {
   try {
     const userPoolId = event.ResourceProperties.userPoolId;
     const hostedUIProviderMeta = JSON.parse(event.ResourceProperties.hostedUIProviderMeta);
+    const hostedUIProviderCreds = JSON.parse(event.ResourceProperties.hostedUIProviderCreds);
+    const stackName = event.ResourceProperties.stackName;
 
-    for (const { ProviderName } of hostedUIProviderMeta) {
-      try {
-        const params = { ProviderName, UserPoolId: userPoolId };
-        await identity.deleteIdentityProvider(params).promise();
-      } catch (e) {
-        if (!e?.code?.toString()?.match(/NotFoundException/)) {
-          // bubble up to outer catch.
+    for (const providerMeta of hostedUIProviderMeta) {
+      const providerCreds = hostedUIProviderCreds.find(({ ProviderName }) => providerMeta.ProviderName === ProviderName);
+      if (providerMeta.ProviderName === 'SignInWithApple' && !providerCreds.private_key && event.RequestType === 'Update') {
+        try {
+          const params = {
+            UserPoolId: userPoolId,
+            ProviderName: providerMeta.ProviderName,
+            AttributeMapping: providerMeta.AttributeMapping,
+          };
+
+          if (providerCreds.client_id && providerCreds.team_id && providerCreds.key_id && providerCreds.private_key) {
+            params.ProviderDetails = {
+              client_id: providerCreds.client_id,
+              team_id: providerCreds.team_id,
+              key_id: providerCreds.key_id,
+              private_key: providerCreds.private_key,
+              authorize_scopes: providerMeta.authorize_scopes,
+            };
+          }
+
```

Comment:
I need to add 1 more migration boolean in the stack builder so that this lambda doesn't get deleted before the `SignInWithApple` is created with CFN